api/src/api/admin.js

```diff
@@ -166,17 +166,141 @@ router.post(
   }),
 );
 
+router.put(
+  '/categories/:id',
+  errorWrap(async (req, res) => {
+    const { id } = req.params;
+    const updatedCategory = await Category.findByIdAndUpdate(id, req.body, {
+      new: true,
+      runValidators: true,
+    });
+
+    const updatedResources = await Resource.updateMany(
+      { category: req.body.currentName },
+      { $set: { 'category.$': req.body.newName } },
+    );
+
+    res.json({
+      code: 200,
+      message: `Successfully updated category ${id}`,
+      success: true,
+      result: {
+        updatedCategory,
+        updatedResources,
+      },
+    });
+  }),
+);
+
 // Delete a category
 router.delete(
   '/categories/:id',
   errorWrap(async (req, res) => {
     const { id } = req.params;
     await Category.findByIdAndDelete(id);
+
+    const updatedResources = await Resource.updateMany(
+      { category: req.body.currentName },
```

Comment:
I think we can just name this `categoryName` or something, `currentName` doesn't make too much sense in this context